dotCMS/src/main/java/com/dotmarketing/quartz/job/DeleteUserJob.java

```diff
@@ -34,47 +40,41 @@ public DeleteUserJob() {
+    public static void triggerDeleteUserJob(final User userToDelete, final User replacementUser, final User user,
+                                           final boolean respectFrontEndRoles) {
 
-        String randomID = UUID.randomUUID().toString();
+        final Map<String, Serializable> nextExecutionData = ImmutableMap
+                .of("userToDelete", userToDelete,
+                        "replacementUser", replacementUser,
+                        "user", user,
+                        "respectFrontEndRoles", respectFrontEndRoles);
 
-        JobDetail jd = new JobDetail("DeleteUserJob-" + randomID, "delete_user_jobs", DeleteUserJob.class);
-        jd.setJobDataMap(dataMap);
-        jd.setDurability(false);
-        jd.setVolatility(false);
-        jd.setRequestsRecovery(true);
-
-        long startTime = System.currentTimeMillis();
-        SimpleTrigger trigger = new SimpleTrigger("deleteUserTrigger-" + randomID, "delete_user_triggers",
-            new Date(startTime));
 
         try {
-            Scheduler sched = QuartzUtils.getSequentialScheduler();
-            UserAPI userAPI = APILocator.getUserAPI();
-            NotificationAPI notAPI = APILocator.getNotificationAPI();
+            final UserAPI userAPI = APILocator.getUserAPI();
+            final NotificationAPI notAPI = APILocator.getNotificationAPI();
 
-            String deleteInProgress = MessageFormat.format(LanguageUtil.get(user,
+            final String deleteInProgress = MessageFormat.format(LanguageUtil.get(user,
                 "com.dotmarketing.business.UserAPI.delete.inProgress"),
                 userToDelete.getUserId() + "/" + userToDelete.getFullName());
 
-            synchronized (userToDelete.getUserId().intern()) {
-                User freshUser = userAPI.loadUserById(userToDelete.getUserId());
-                if(! freshUser.isDeleteInProgress()) {
+            final IdentifierStripedLock lockManager = DotConcurrentFactory.getInstance().getIdentifierStripedLock();
+            lockManager.tryLock(userToDelete.getUserId(), () -> {
+                final User freshUser = userAPI.loadUserById(userToDelete.getUserId());
+                if (!freshUser.isDeleteInProgress()) {
                     userAPI.markToDelete(userToDelete);
-                    sched.scheduleJob(jd, trigger);
+                    try {
+                        DotStatefulJob.enqueueTrigger(nextExecutionData, DeleteUserJob.class);
+                    }catch (ParseException | SchedulerException | ClassNotFoundException e){
+                        Logger.error(DeleteUserJob.class, "Error scheduling DeleteUserJob", e);
+                        throw new DotRuntimeException("Error scheduling DeleteUserJob", e);
+                    }
                 } else {
                     notAPI.info(deleteInProgress, user.getUserId());
                 }
-            }
+            });
 
-
-        } catch (SchedulerException e) {
+        } catch (Throwable e) {
```

Comment:
Codacy found an issue: [A catch statement should never catch throwable since it includes errors.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077927)